public/app/features/variables/pickers/OptionsPicker/reducer.ts

```diff
@@ -178,20 +187,39 @@ const optionsPickerSlice = createSlice({
       };
     },
     toggleAllOptions: (state, action: PayloadAction): OptionsPickerState => {
-      if (state.selectedValues.length > 0) {
+      // Check if 'All' option is configured and if it's selected
+      const isAllSelected = state.selectedValues.find((option) => option.value === ALL_VARIABLE_VALUE);
+      const isAllConfigured = state.options.find((option) => option.value === ALL_VARIABLE_VALUE);
+
+      // If 'All' option is not selected but some options are, clear all options and select 'All'
+      if (state.selectedValues.length > 0 && isAllConfigured && !isAllSelected) {
         state.selectedValues = [];
+        const allOptionState = state.options.find((option) => option.value === ALL_VARIABLE_VALUE);
+        state.selectedValues.push({
+          text: allOptionState?.text ?? 'All',
+          value: allOptionState?.value ?? ALL_VARIABLE_VALUE,
+          selected: true,
+        });
```

Comment:
If you already found the All option based on option.value === ALL_VARIABLE_VALUE, that means the value is not null and that is ALL_VARIABLE_VALUE. (using the `isAllConfigured` that is checked in the if)

```suggestion
        state.selectedValues.push({
          text: isAllConfigured.text ?? 'All',
          value: isAllConfigured.value,
          selected: true,
        });
```

Reply:
I put `isAllConfigured` but it would be `allOptionState` name. However, that const is from the .find before the if.

```
const allOptionState = state.options.find((option) => option.value === ALL_VARIABLE_VALUE);
if (state.selectedValues.length > 0 && isAllConfigured && !isAllSelected) {
  // the code mentioned
}
```